lib/src/models/subreddit_moderation.dart

```diff
@@ -0,0 +1,647 @@
+}
+
+ModeratorAction buildModeratorAction(Reddit reddit, Map data) =>
+    new ModeratorAction._(reddit, data);
+
+/// Represents an action taken by a moderator.
+class ModeratorAction {
+  final Reddit _reddit;
+  final Map data;
+  ModeratorAction._(this._reddit, this.data);
+
+  ModeratorActionType get action =>
+      _stringToModeratorActionType(data['action']);
+
+  RedditorRef get mod => _reddit.redditor(data['mod']);
+
+  SubredditRef get subreddit => _reddit.subreddit(data['subreddit']);
+
+  @override
+  String toString() {
+    if (data != null) {
+      final encoder = new JsonEncoder.withIndent('  ');
+      return encoder.convert(data);
+    }
+    return 'null';
+  }
+}
+
+/// Provides a set of moderation functions to a [Subreddit].
+class SubredditModeration {
+  static final _subredditRegExp = new RegExp(r'{subreddit}');
+  final SubredditRef _subreddit;
+  SubredditModeration(this._subreddit);
+
+  Map _buildOnlyMap(SubredditModerationContentTypeFilter only) {
+    final params = {};
+    if (only != null) {
+      var _only;
+      if (only == SubredditModerationContentTypeFilter.submissionsOnly) {
+        _only = 'links';
+      } else {
+        _only = 'comments';
+      }
+      params['only'] = _only;
+    }
+    return params;
+  }
+
+  Stream _subredditModerationListingGeneratorHelper(
+          String api, SubredditModerationContentTypeFilter only, {int limit}) =>
+      ListingGenerator.generator(_subreddit.reddit, _formatApiPath(api),
+          params: _buildOnlyMap(only), limit: limit);
+
+  String _formatApiPath(String api) =>
+      apiPath[api].replaceAll(_subredditRegExp, _subreddit.displayName);
+
+  /// Accept an invitation to moderate the community.
+  Future acceptInvite() async {
+    final url = apiPath['accept_mod_invite']
+        .replaceAll(_subredditRegExp, _subreddit.displayName);
+    return _subreddit.reddit.post(url, {});
+  }
+
+  /// Returns a [Stream<UserContent>] of edited [Comment]s and [Submission]s.
+  ///
+  /// If `only` is provided, only [Comment]s or [Submission]s will be returned
+  /// exclusively.
+  Stream<UserContent> edited(
+          {SubredditModerationContentTypeFilter only, int limit}) =>
+      _subredditModerationListingGeneratorHelper('about_edited', only,
+          limit: limit);
+
+  /// Returns a [Stream<Message>] of moderator messages.
+  Stream<Message> inbox({int limit}) =>
+      _subredditModerationListingGeneratorHelper('moderator_messages', null,
+          limit: limit);
+
+  /// Returns a [Stream<ModeratorAction>] of moderator log entries.
+  ///
+  /// If `mod` is provided, only log entries for the specified moderator(s)
+  /// will be returned. `mod` can be a [RedditorRef], [List<RedditorRef>], or
+  /// a [String] of comma-separated moderator names. If `type` is provided,
+  /// only [ModeratorAction]s of that [ModeratorActionType] will be returned.
+  Stream<ModeratorAction> log(
+      {/* RedditorRef, List<RedditorRef>, String */ mod,
```

Comment:
Is there some way to bring in a new [Mod] Type that can be extended to encapsulate these three types, I just don't like the type checking going on in the conditional below, but I can't think of a way to fix it either 😞 .

Reply:
I don't believe so, not in a way that would allow for the argument to be of any of those types.